File: mRemoteV1/Tools/PuttyProcessController.vb

Imports mRemote3G.App.Info

Namespace Tools
    Public Class PuttyProcessController
        Inherits ProcessController

        Public Overloads Function Start(Optional ByVal arguments As CommandLineArguments = Nothing) As Boolean
            Dim filename As String
            If My.Settings.UseCustomPuttyPath Then
                filename = My.Settings.CustomPuttyPath
            Else
                filename = General.PuttyPath
            End If
            Return Start(filename, arguments)
        End Function
    End Class
End Namespace

File: mRemoteV1/CS/Tools/PuttyProcessController.cs
using System.Collections.Generic;
using System;
using AxWFICALib;
using System.Drawing;
using System.Diagnostics;
using System.Data;
using AxMSTSCLib;
using Microsoft.VisualBasic;
using System.Collections;
using System.Windows.Forms;
using mRemoteNG.My;


namespace mRemoteNG.Tools
{
	public class PuttyProcessController : ProcessController
	{
		public bool Start(CommandLineArguments arguments = null)
		{
			string filename = "";
			if (My.Settings.Default.UseCustomPuttyPath)
			{
				filename = My.Settings.Default.CustomPuttyPath;
			}
			else
			{
				filename = App.Info.General.PuttyPath;
			}
			return Start(filename, arguments);
		}
	}
}
